test/router/V4Router.t.sol

```diff
@@ -145,45 +98,125 @@ contract V4RouterTest is RoutingTestHelpers {
         tokenPath.push(currency3);
         IV4Router.ExactInputParams memory params = _getExactInputParams(tokenPath, amountIn);
 
-        uint256 prevBalance0 = currency0.balanceOfSelf();
-        uint256 prevBalance3 = currency3.balanceOfSelf();
+        plan = plan.add(Actions.SWAP_EXACT_IN, abi.encode(params));
+
+        _finalizeExecuteAndCheckSwap(currency0, currency3, amountIn, expectedAmountOut);
+
+        // check intermediate token balances
+        assertEq(currency1.balanceOf(address(router)), 0);
+        assertEq(currency2.balanceOf(address(router)), 0);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                ETH -> ERC20 and ERC20 -> ETH EXACT INPUT
+    //////////////////////////////////////////////////////////////*/
+
+    function test_nativeIn_swapExactInputSingle() public {
+        uint256 amountIn = 1 ether;
+        uint256 expectedAmountOut = 992054607780215625;
+
```

Comment:
```suggestion
// native input means we need zeroForOne=true
```

Reply:
native token is always currency0.. its just a little confusing reading these tests bc i think the nativeKey is inialized with native, and also the global currency0, which becomes currency1 in the native pool

maybe where you have currency0 you can use nativeKey.currency1? idk or a comment somewhere